packages/block-editor/src/components/colors-gradients/control.js

```diff
@@ -78,24 +92,16 @@ function ColorGradientControlInner( {
 						</legend>
 					) }
 					{ canChooseAColor && canChooseAGradient && (
-						<ToggleGroupControl
-							value={ currentTab }
-							onChange={ setCurrentTab }
-							label={ __( 'Select color type' ) }
-							hideLabelFromVision
-							isBlock
+						<TabPanel
+							className="block-editor-color-gradient-control__tabs"
+							onSelect={ setCurrentTab }
+							tabs={ TABS_SETTINGS }
 						>
-							<ToggleGroupControlOption
-								value="color"
-								label={ __( 'Solid' ) }
-							/>
-							<ToggleGroupControlOption
-								value="gradient"
-								label={ __( 'Gradient' ) }
-							/>
-						</ToggleGroupControl>
+						{ ( tab ) => <p className="screen-reader-text">Selected tab: { tab.title }</p> }
```

Comment:
I believe what needs to happen here is to provide a function that receives the currently selected tab as an argument and render the tab panel for it. Basically the `TabPanel` is going to manage our `currentTab` state from now on. That is how it handles the a11y concerns for us.

Since it looks like we're going to use the tab panel content outside of a TabPanel context as well, perhaps we could do some light refactoring along these lines:

```jsx
const tabPanels = {
  [TAB_COLOR.value]: <ColorPalette foo={bar} />;
  [TAB_GRADIENT.value]: <GradientPicker foo={bar} />;
};
```

```jsx
					{ canChooseAColor && canChooseAGradient && (
						/* Note that we don't need the `onSelect` anymore */
						<TabPanel
							className="block-editor-color-gradient-control__tabs"
							tabs={ TABS_SETTINGS }
						>
						  { ( tab ) => tabPanels[ tab ]; }
						</TabPanel>
					) }
					{ ! canChooseAGradient && tabPanels[ TAB_COLOR.value ] }
					{ ! canChooseAColor && tabPanels[ TAB_GRADIENT.value ] }
```

Reply:
Oh yeah, that makes sense :)

I tried to implement this as you suggested but my designer brain has obviously messed something up because the panels aren't rendering for me 🤔 I don't suppose you could take a look?

Reply:
Ah yes, of course! I wrote that code snippet without running it so something is likely wrong 😂 I'll rework the code and push it directly to this PR branch then 👍

Reply:
Ok, done ✅ Passing the torch back to you!